### PR TITLE
Event based webhooks where filtering out all webhooks if the event envs was empty 

### PR DESCRIPTION
**Pull request by gazzdingo**
### Features and Changes

- Closes **(add link to issue here)**

### CODE REVIEW

**Review by romain-growthbook — APPROVED** [review]
Thanks. I'm gonna add a test and merge.

**Comment by jdorn** [comment]
Was this already fixed in https://github.com/growthbook/growthbook/pull/2324/files?

**Comment by romain-growthbook** [comment]
> Was this already fixed in https://github.com/growthbook/growthbook/pull/2324/files?

Yes, I saw it afterward. I'd like to merge this one to make sure we have a regression test as well.

File: packages/back-end/src/events/handlers/utils.ts [hunk 1]
```diff
@@ -131,20 +131,23 @@ export const filterEventForEnvironments = ({
...
+export const filterFeatureUpdatedNotificationEventForEnvironments = ({
   featureEvent,
   environments,
 }: {
   featureEvent: FeatureUpdatedNotificationEvent;
   environments: string[];
 }): boolean => {
-  const { previous, current } = featureEvent.data;
+  // if the environments are not specified, notify for all environments
+  if (environments.length === 0) {
```
[the remark this answers is not in the document]
Comment (romain-growthbook) [reply packages/back-end/src/events/handlers/utils.ts:142]: In general I like to maintain strict boundaries and levarage the type to make sure we don't end up with nullable variables everywhere. Here, the type is strict `string[]` and it looks like the call site is properly protected:

```typescript
      return (
        (await getAllEventWebHooksForEvent({
          organizationId: event.organizationId,
          eventName: event.data.event,
          enabled: true,
          tags,
          projects,
        })) || []
      ).filter(({ environments = [] }) =>
        filterEventForEnvironments({ event: event.data, environments })
      );
```

[the remark this answers is not in the document]
Comment (romain-growthbook) [reply packages/back-end/src/events/handlers/utils.ts:142]: The return type of the `getAllEventWebHooksForEvent` function also seems correct:

```typescript
export interface EventWebHookInterface {
  id: string;
  organizationId: string;
  name: string;
  dateCreated: Date;
  dateUpdated: Date;
  enabled: boolean;
  events: NotificationEventName[];
  url: string;
  signingKey: string;
  // Last state
  lastRunAt: Date | null;
  lastState: "none" | "success" | "error";
  lastResponseBody: string | null;
  tags?: string[];
  environments?: string[];
  projects?: string[];
  headers?: Record<string, string>;
  payloadType?: EventWebHookPayloadType;
  method?: EventWebHookMethod;
}
```

**Comment by gazzdingo** [comment]
Yeah Bryce and I were both working on it last night and he bet me to merging lol

**Comment by romain-growthbook** [comment]
> Yeah Bryce and I were both working on it last night and he bet me to merging lol

Yeah I saw that. I want to merge this one though, I added a regression test so that we make sure this does not happen again..

**Comment by gazzdingo** [comment]
We should also add the copy to tags since they are the same can you add that in a new pr?

**Comment by gazzdingo** [comment]
We also should add test cases for all the different filters that are there not just the env

**Pull request merged**